File: DSA/ManasMaheshwari_590015617/Day_4/Question2.java
// Question: Given an integer array and an integer k, rotate the array to the right by k steps.
// Time Complexity: O(N)
// Space Complexity: O(N)

class Question2 {
    public static void main(String args[]) {
        int array[] = {1, 2, 3, 4};
        int count = array.length;
        int values_rotated = 3;
        
        values_rotated = values_rotated % count;
        int new_array[] = new int[count];
        
        for (int i = 0; i < count; i++) {
            new_array[(i + values_rotated) % count] = array[i];
        }
        
        for (int i = 0; i < count; i++) {
            System.out.print(new_array[i] + " ");
        }
        System.out.println();
    }
}

File: DSA/krishna_590022060/day_4/question2.c
//Given an integer array and an integer k, rotate the array to the right by k steps.
#include<stdio.h>
void rotate(int arr[],int n,int k){
    k=k%n;
    int temp[n];
    for(int i=0;i<n;i++){
        temp[(i+k)%n]=arr[i];
    }
    for(int i=0;i<n;i++){
        arr[i]=temp[i];
    }

}
int main(){
    int arr[]={1,2,3,4,5,6,7};
    int n=sizeof(arr)/sizeof(arr[0]);
    int k=3;
    rotate(arr,n,k);
    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);
    }
    return 0;
}